### ISSUE TITLE
service-worker plugin triggers a rollup error

### ISSUE THREAD

**Issue by AbelVM** [preactjs/wmr#794]
**Describe the bug**

I'm getting the following error in dev console

```javascript
_wmr.js:3 [wmr]  Connected to server.

_wmr.js:111 Unexpected token (Note that you need plugins to import files that are not JavaScript)
undefined
handleMessage @ _wmr.js:111

main.js:23 GET http://localhost:8080/@sw/service-worker.js net::ERR_ABORTED 500 (Internal Server Error)
```
Being `main.js:23`:

```javascript
import swURL from 'sw:../service-worker.js';
```

At the terminal:
```bash
500 /@sw/service-worker.js - Unexpected token (Note that you need plugins to import files that are not JavaScript)

  at Error: Unexpected token (:-1:-1)
  at error (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:8078:30)
  at Module.error (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:9977:16)
  at Module.tryParse (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:10380:25)
  at Module.setSource (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:10283:24)
  at ModuleLoader.addModuleSource (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:19626:20)
  at processTicksAndRejections (internal/process/task_queues.js:95:5)
  at async ModuleLoader.fetchModule (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:19682:9)
  at async Promise.all (:-1:-1)
  at async ModuleLoader.fetchStaticDependencies (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:19708:34)
  at async Promise.all (:-1:-1)
```

On the other hand, I need `wmr.config` to be `.mjs` not `.js` as in the example code, because it fails with

```bash
(node:37119) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/home/abel/Projects/PWA/wmr.config.js:16
import swPlugin from '@wmrjs/service-worker';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:988:16)
    at Module._compile (internal/modules/cjs/loader.js:1036:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at ModuleWrap.<anonymous> (internal/modules/esm/translators.js:199:29)
    at ModuleJob.run (internal/modules/esm/module_job.js:170:25)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async DZe (/home/abel/Projects/PWA/node_modules/wmr/wmr.cjs:2:2557856)
    at async lct (/home/abel/Projects/PWA/node_modules/wmr/wmr.cjs:2:2908310)

Error: Failed to load wmr.config.js
SyntaxError: Cannot use import statement outside a module
TypeError [ERR_INVALID_ARG_TYPE]: The "id" argument must be of type string. Received an instance of URL

    at DZe (/home/abel/Projects/PWA/node_modules/wmr/wmr.cjs:2:2558011)
    at async lct (/home/abel/Projects/PWA/node_modules/wmr/wmr.cjs:2:2908310)
    at async cct (/home/abel/Projects/PWA/node_modules/wmr/wmr.cjs:2:2907826)
```` 

**To Reproduce**

Using latest versions of both `wmr` and `@wmrjs/service-worker,` with a minimal testcase following the instructions at https://github.com/preactjs/wmr/tree/main/packages/sw-plugin or just the example at https://github.com/preactjs/wmr/tree/main/packages/sw-plugin/example

**Expected behavior**
Just work 

**Bug occurs with:**

- [x] `wmr` or `wmr start` (development)
- [ ] `wmr build` (production)
- [ ] `wmr serve`

**Desktop (please complete the following information):**

- OS: Ubuntu 20.04
- Browser: Chrome
- Node Version: 14.17.5
- WMR Version: 3.5.0
- WMR service-workers plugin: 0.3.0

**Additional context**

**Comment by AbelVM** [comment]
I have built another test case, different error message

`wmr.config.mjs`
```javascript
import swPlugin from '@wmrjs/service-worker';
export default function (options) {
  swPlugin(options);
}
```
`public/index.html`
```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>TEST</title>
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <link rel="preload" as="script" href="js/main.js">
</head>
<body>
  <script type="module" src="js/main.js"></script>
</body>
</html>
```
`public/js/main.js`
```javascript
import swURL from 'sw:../service-worker.js';
if ('serviceWorker' in navigator) {
 window.addEventListener('load', async () => {
      navigator.serviceWorker.register(swURL);
  });
} 
```
`public/service-worker.js`
```javascript
// lots of workbox stuff
```

And running WMR I get a similar message in dev console
```javascript
_wmr.js:3 [wmr]  Connected to server.
_wmr.js:111 Could not resolve entry module (service-worker.js).
undefined
handleMessage @ _wmr.js:111
service-worker.js:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
```
But a different one in terminal
```bash
500 /@sw/service-worker.js - Could not resolve entry module (service-worker.js).

  at error (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:8078:30)
  at ModuleLoader.loadEntryModule (file:///home/abel/Projects/PWA/node_modules/rollup/dist/es/shared/rollup.js:19782:20)
  at processTicksAndRejections (internal/process/task_queues.js:95:5)
  at async Promise.all (:-1:-1)
```
I have triple-checked the relative paths, and they look ok, but there is no way to load the service worker file. 

The rendered `main.js` looks like:

```javascript
import swURL from '/@sw/service-worker.js';
// Register the service worker
if ('serviceWorker' in navigator) {
  navigator.serviceWorker.register(swURL)
} 
import { createHotContext as $w_h$ } from '/_wmr.js'; $w_h$(import.meta.url);
```
As expected, there's a `/_wmr.js` served by WMR, but no `/@sw/service-worker.js` to be found :thinking:

**Comment by AbelVM** [comment]
Looks like this issue is related with https://github.com/preactjs/wmr/issues/773

**Comment by developit** [comment]
This is fixed by #796. You can copy the source of that plugin file into your config to test it out locally before its released.
